Recruit-Training/Advanced-Recruit-Training/Telemetry-Teaser/src/app/app.component.html

```diff
@@ -16,12 +16,12 @@ <h1> Elysia Range Simulator </h1>
       <br>
       <button type="submit" (click)="calculateRange()">Calculate</button>
       <br>
-      <label *ngIf="(speedInput.invalid && (speedInput.dirty || speedInput.touched)) && this.buttonPress">
+      <label *ngIf="(speedInput.invalid && (speedInput.dirty || !speedInput.touched)) && this.buttonPress">
```

Comment:
This fixes the problem where the error doesn't show when the inputs are untouched when the form is submitted but it introduces a new problem. Now if the user clicks on the input field but leaves it blank and submits the form, no error appears. 
There are probably a few ways to fix this but here is a hint for one way that follows what you did in this commit. Consider what your `ngIf` statement is saying. What can you add or change to check all cases?